bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/command/input/Input.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pjlinkdevice.internal.device.command.input;
+
+import org.openhab.binding.pjlinkdevice.internal.device.command.ResponseException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.regex.Pattern;
+
+/**
+ * Describes an A/V source that can be selected on the PJLink device.
+ *
+ * @author Nils Schnabel - Initial contribution
+ */
+@NonNullByDefault
+public class Input {
+
+  private static final Pattern INPUT_NUMBER_PATTERN = Pattern.compile("[0-9A-Z]");
+
+  enum InputType {
+    RGB("RGB", '1'),
+    VIDEO("Video", '2'),
+    DIGITAL("Digital", '3'),
+    STORAGE("Storage", '4'),
+    NETWORK("Network", '5');
+
+    private String text;
+    private char code;
+
+    private InputType(String text, char code) {
+      this.text = text;
+      this.code = code;
+    }
+
+    public String getText() {
+      return this.text;
+    }
+
+    public static InputType parseString(String value) throws ResponseException {
+      for (InputType result : InputType.values()) {
+        if (result.code == value.charAt(0)) {
+          return result;
+        }
+      }
+
+      throw new ResponseException("Unknown input channel type: " + value);
+    }
+  }
+
+  private String value;
+
+  public Input(String value) throws ResponseException {
+    this.value = value;
+    validate();
+  }
+
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + value.hashCode();
+    return result;
+  }
+
+  @Override
+  public boolean equals(@Nullable Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (getClass() != obj.getClass()) {
```

Comment:
```suggestion
    if (obj instanceof Input) {
```

since you cast to `Input` later

Reply:
instanceof would also match subclasses. As this method checks for equality, we can already stop if the two objects are not of the exact same class.

Btw. this is generated code made by the 'generate equals() and hashCode()' wizard of Eclipse, and I think they are typically exactly what is needed :)